src/mpid/ch3/channels/nemesis/include/mpid_nem_defs.h

```diff
@@ -34,22 +34,22 @@ typedef MPI_Aint MPID_nem_addr_t;
 extern  char *MPID_nem_asymm_base_addr;
 
 #define MPID_NEM_REL_NULL (0x0)
-#define MPID_NEM_IS_REL_NULL(rel_ptr) (OPA_load_ptr(&(rel_ptr).p) == MPID_NEM_REL_NULL)
-#define MPID_NEM_SET_REL_NULL(rel_ptr) (OPA_store_ptr(&((rel_ptr).p), MPID_NEM_REL_NULL))
+#define MPID_NEM_IS_REL_NULL(rel_ptr) (MPL_atomic_relaxed_load_ptr(&(rel_ptr).p) == MPID_NEM_REL_NULL)
```

Comment:
Do we want to use stronger semantics anyway? `relaxed` doesn't guarantee ordering, right? I am having a hard time to imagine how that can be useful.

Reply:
If the `rel_ptr` has been already acquired, we don't need an acquire load here.
We need to examine all the call sites. The same thing goes to the other macros.

Reply:
`OPA_load_ptr` and `OPA_store_ptr` were using relaxed semantics to begin with.  Looking at the nemesis implementation it appears the barrier usage is correct.  I recommend against sprinkling barriers around just-in-case, they can adversely impact performance as well as mask true bugs.  

`relaxed` ordering is useful here because you get the benefit of the load being 'visible' to the outside and atomic, even though it does not require any order. As @shintaro-iwasaki discussed with me in #4222, not tagging the proper loads/stores as atomic allows the compiler to do unintended optimizations.